packages/core/src/utils/tenant.ts

```diff
@@ -74,7 +77,17 @@ const getTenantIdFromCustomDomain = async (
   return domain?.tenantId;
 };
 
-export const getTenantId = async (url: URL) => {
+/**
+ * Get tenant ID from the current request's URL.
+ *
+ * @param url The current request's URL
+ * @param skipCustomDomain Indicating whether to skip looking for custom domain
+ * @returns tenantId or undefined
+ */
+export const getTenantId = async (
+  url: URL,
+  skipCustomDomain?: boolean
+): Promise<string | undefined> => {
```

Comment:
instead of having two separate functions, how about:

```suggestion
export const getTenantId = async (
  url: URL,
  skipCustomDomain?: boolean
): Promise<[tenantId: string, isCustomDomain: boolean] | undefined> => {
```

Reply:
updated